example-apps/internal-knowledge-search/app-ui/src/App.tsx

```diff
@@ -21,8 +21,8 @@ function App() {
                         <SidebarMenu/>
                         <main className="flex-1 p-6">
                             <Routes>
-                                <Route path="/" element={<SearchPage/>}/>
-                                <Route path="/settings" element={<SettingsPage/>}/>
+                                <Route path="/" element={<SettingsPage/>}/>
+                                <Route path="/search" element={<SearchPage/>}/>
```

Comment:
This is a hack. Since I'm generating api keys dynamically now, but on settings page load, I force you to start at the settings page. 

It would be better to have the API Key generation logic be resuable between both the search and settings pages, but it wasn't clear to me what the right pattern/location for the code would be. I'd welcome guidance and/or examples on the right way to structure "library" code in react.